### DISCUSSION TITLE
GitHub Action self-hosted runner fails configuration for private repository

### DISCUSSION THREAD

**Discussion by frostebite** [community/community#25567] [Actions]
<p>I’m trying to setup a GitHub action self-hosted runner on a GCP VM running windows. I’m able to get up to the configuration step. </p>
<p>When I run the config script I see the following:</p>
<p><a href="http://prntscr.com/pxi3v8" rel="nofollow noopener">http://prntscr.com/pxi3v8</a></p>
<p>Note: when I try and enter the repo and token manually that also fails.</p>
<p>I suspect this is because the repository I’m trying to target is private, but this is the first self-hosted runner I have tried to setup so I’m unsure. I did run this in a powershell instance with admin privileges. </p>

**Comment by bubthegreat** [comment]
<p>I’m not having the same problem with private repositories, FYI - but I did have trouble until I did it all in a single line command that’s given when clicking the add-runner button.</p>

**Comment by frostebite** [comment]
<p>Hey, appreciate the info. </p>
<p>I ran the commands from the “Add new runner” dialog in order, in powershell without closing and reopening the command line window. I tried this running powershell in admin, and I tried it without also. So I’m not sure if this is the solution.</p>

**Comment by frostebite** [comment, marked as answer]
<p>I simply kept trying the exact same way, and a few hours later it managed to print “Connected to Github successfully” and asked me to enter a name for the runner.</p>
<p>However it then printed </p>
<p>“The SSL connection could not be established, see inner exception.”</p>
<p>I tried a few more times and it then succeeded.</p>
<p>*shrugs* it is a beta I suppose …</p>

**Comment by Paperone80** [comment]
<p>Hi,</p>
<p>I am on Ubuntu 16.04. This solved it for me: </p>
<pre><code>$ sudo update-ca-certificates
Updating certificates in /etc/ssl/certs...
...
done.
</code></pre>
<p>Use the certs folder in front of .configure in env SSL_CERT_DIR=</p>
<pre><code>$ SSL_CERT_DIR=/etc/ssl/certs ./config.sh --url https://github.com/...
</code></pre>
<p>You should see:</p>
<pre><code>--------------------------------------------------------------------------------
| ____ _ _ _ _ _ _ _ _ |
| / ___(_) |_| | | |_ _| |__ / \ ___| |_(_)___ _ _____ |
| | | _| | __| |_| | | | | '_ \ / _ \ /__ | __| |/ _ \| '_ \/__ | |
| | |_| | | |_| _ | |_| | |_) | / ___\ (__| |_| | (_) | | | \__ \ |
| \ ____|_|\__ |_| |_|\ __,_|_.__ / /_/ \_\ ___|\__ |_|\ ___/|_| |_|___ / |
| |
| Self-hosted runner registration |
| |
--------------------------------------------------------------------------------

# Authentication


√ Connected to GitHub

# Runner Registration

Enter the name of runner: [press Enter for ${hostname -s}]

√ Runner successfully added
√ Runner connection is good

# Runner settings

Enter name of work folder: [press Enter for _work]

√ Settings Saved.
</code></pre>
<p>Good luck!</p>
